src/equations/shallow_water_2d.jl

```diff
@@ -6,7 +6,7 @@
 
 
 @doc raw"""
-    ShallowWaterEquations2D(gravity, H0)
+    ShallowWaterEquations2D(gravity, H0, threshold_limiter, threshold_wet)
```

Comment:
It is also used to implement an initial shift before returning the value out of each initial condition function. Therefore, I store it in the equations struct to keep it al together

Reply:
But it's also the same which should be used by the limiter? In this case, the limiter doesn't need such a threshold anymore, does it?

Reply:
I think I do not quit understand your point...
Do you mean that there could be a different way to store the threshold or that it would be obsolete in one part? In fact, both applications of the limiter are needed.

Reply:
As far as I understand, you
- use the `threshold` in the `equations` to "fix" the initial condition
- use the `threshold` in the `limiter` for the positivity-preserving limiting during the simulation
- basically assume/require that both `threshold`s are the same

Because of the last aspect, I wonder whether it makes sense to store the (necessarily same?) value twice. If they are required/assumed to be the same, this creates a trap.

Reply:
I extended the SWE struct just because I do not want the store the `threshold_limiter` twice and because it is used twice. That is why I plugged it into the struct and get access to it when needed. And the `threshold_wet` is used in another way with another default value

Reply:
But you do store it twice - in the limiter and the equations?

Reply:
At least it is passed to the limiter on the ‚elixir-level‘, yes

Reply:
I hadn't thought about the two copies of the `threshold_limiter`.  It is convenient for the `equations` to know about the `threshold_limiter` at several places, especially in the initial conditions. This avoids hard coding values for each `elixir` in the initialization phase. To avoid this "trap" would it be sufficient to add a check that the value passed to `thresholds` matches the existing value in the `equations`?

Reply:
Wouldn't it be possible for the limiter to use the value from the equations? If it doesn't make sense that both values might differ, then IMHO only one value should be stored and used everywhere.

Reply:
> Wouldn't it be possible for the limiter to use the value from the equations

That's exactly what I would propose. If you need such a threshold in the equations, you shouldn't duplicate it in the limiter.

Reply:
So it would be ok to delete the threshold from the `PositivityPreservingLimiterShallowWater` struct and the complete wrapper logic. Therefore, it would not be passed it to the method. But I get access to it within the method by the equations struct. 
I just wanted to make sure, is this what you propose?

Reply:
Yes, because of the following reasoning - correct me if I'm wrong somewhere.
- You need to access the threshold used by limiter also from the equations for setting ICs etc.
- You will get into trouble if the thresholds from the equations and the limiter do not match

Hence, I would
- remove the threshold from the new limiter and restrict it to keeping positivity of the water height - since this is the only thing you want it to do
- grab the threshold from the equations wherever necessary

Reply:
Ok, I totally see the point. I just wanted to be sure that I understand your suggestion right. I will update this

Reply:
@andrewwinters5000 @sloede @ranocha I just pushed the changes in the code regarding the SWE limiter.

Reply:
Done. Now the PositivityLimiter uses the value of this `threshold_limiter` from the ShallowWater struct to avoid a user accidentally setting two limiter values.